hypothesis-python/tests/array_api/test_arrays.py

```diff
@@ -0,0 +1,506 @@
+        xps.unsigned_integer_dtypes(),
+        xps.floating_dtypes(),
+    ],
+)
+@given(st.data())
+def test_draw_arrays_from_dtype_strategies(strat, data):
+    """Draw arrays from dtype strategies."""
+    x = data.draw(xps.arrays(strat, ()))
+    assert_array_namespace(x)
+
+
+@given(st.lists(st.sampled_from(DTYPE_NAMES), min_size=1, unique=True), st.data())
+def test_draw_arrays_from_dtype_name_strategies(names, data):
+    """Draw arrays from dtype name strategies."""
+    names_strategy = st.sampled_from(names)
+    x = data.draw(xps.arrays(names_strategy, ()))
+    assert_array_namespace(x)
+
+
+@given(xps.arrays(xp.int8, xps.array_shapes()))
+def test_generate_arrays_from_shapes_strategy(x):
+    """Generate arrays from shapes strategy."""
+    assert_array_namespace(x)
+
+
+@given(xps.arrays(xp.int8, st.integers(0, 100)))
+def test_generate_arrays_from_integers_strategy_as_shape(x):
+    """Generate arrays from integers strategy as shapes strategy."""
+    assert_array_namespace(x)
+
+
+@given(xps.arrays(xp.int8, ()))
+def test_generate_arrays_from_zero_dimensions(x):
+    """Generate arrays from empty shape."""
+    assert x.shape == ()
+    assert_array_namespace(x)
+
+
+@given(xps.arrays(xp.int8, (1, 0, 1)))
+def test_handle_zero_dimensions(x):
+    """Generate arrays from empty shape."""
+    assert x.shape == (1, 0, 1)
+    assert_array_namespace(x)
+
+
+@given(xps.arrays(xp.uint32, (5, 5)))
+def test_generate_arrays_from_unsigned_ints(x):
+    """Generate arrays from unsigned integer dtype."""
+    assert xp.all(x >= 0)
+    assert_array_namespace(x)
+
+
+def test_minimize_arrays_with_default_dtype_shape_strategies():
+    """Strategy with default scalar_dtypes and array_shapes strategies minimize
+    to a boolean 1-dimensional array of size 1."""
+    smallest = minimal(xps.arrays(xps.scalar_dtypes(), xps.array_shapes()))
+    assert smallest.shape == (1,)
+    assert smallest.dtype == xp.bool
+    assert not xp.any(smallest)
+
+
+def test_minimize_arrays_with_0d_shape_strategy():
+    """Strategy with shape strategy that can generate empty tuples minimizes to
+    0d arrays."""
+    smallest = minimal(xps.arrays(xp.int8, xps.array_shapes(min_dims=0)))
+    assert smallest.shape == ()
+
+
+@pytest.mark.parametrize("dtype", NUMERIC_NAMES)
+def test_minimizes_numeric_arrays(dtype):
+    """Strategies with numeric dtypes minimize to zero-filled arrays."""
+    smallest = minimal(xps.arrays(dtype, (2, 2)))
+    assert xp.all(smallest == 0)
+
+
+def test_minimize_large_uint_arrays():
+    """Strategy with uint dtype and largely sized shape minimizes to a good
+    example."""
+    smallest = minimal(
+        xps.arrays(xp.uint8, 100),
+        lambda x: xp.any(x) and not xp.all(x),
+        timeout_after=60,
+    )
+    assert xp.all(xp.logical_or(smallest == 0, smallest == 1))
+    if hasattr(xp, "nonzero"):
+        idx = xp.nonzero(smallest)[0]
+        assert idx.size in (1, smallest.size - 1)
+    else:
+        pytest.skip()
```

Comment:
Tests which depend on optional API would probably be clearer using skip decorators, rather than raising the skip-exception from inside the test:

```python
needs_xp_nonzero = pytest.mark.skipif(not hasattr(xp, "nonzero"), reason="optional API")
...

@needs_xp_nonzero
@given(...)
def test_minimize_large_uint_arrays(...):
    ...
    idx = xp.nonzero(smallest)[0]
    assert idx.size in (1, smallest.size - 1)
```

This is also somewhat faster, since `pytest` can avoid executing these tests at all.

Reply:
Take out the if/else block now that we have the decorator form, as you did for `unique` 😉